sui_core/src/gateway_state.rs

```diff
@@ -573,7 +575,19 @@ where
         tx: Transaction,
     ) -> Result<TransactionResponse, anyhow::Error> {
         let tx_kind = tx.data.kind.clone();
-        let (certificate, effects) = self.execute_transaction_impl(tx).await?;
+        let mut res = self.execute_transaction_impl(tx.clone()).await;
+
+        let mut remaining_retries = MAX_NUM_TX_RETRIES - 1;
```

Comment:
Intuitively RETRY means retry, so it should exclude the first try. Then you don't have to require MAX_NUM_TX_RETRIES to be non-zero.

Reply:
True. I should've said `num_tries`